airflow/models/abstractoperator.py

```diff
@@ -188,6 +188,33 @@ def get_flat_relatives(self, upstream: bool = False) -> Collection[Operator]:
             return set()
         return [dag.task_dict[task_id] for task_id in self.get_flat_relative_ids(upstream=upstream)]
 
+    def get_upstreams_follow_setups(self) -> Iterable[Operator]:
+        """All upstreams and, for each upstream setup, its respective teardowns."""
```

Comment:
Let’s mark these functions with `:meta private:`

Reply:
I thought about it but the whole class is marked private so is it necessary?  If it is necessary, what is the meaning / import of the class being private?

Reply:
Yeah.... the problem with `:meta private:` is it has no effect on IDE behavior.  So while it's technically private, users would have a tough time seeing that.

So. leading underscore is much better, from that perspective.

However, if we use leading underscore, then, wherever we use it, it looks like a bad usage.  Because we're trying to indicate private w.r.t. user, not private w.r.t. the class (which is what underscore means).

and the IDE can't distinguish between user and internal. Wish there was a better solution.  But for now seems meta private is best we can do.